Dialogs/PreferencesDialog.cpp

```diff
@@ -1,22 +1,78 @@
 #include "PreferencesDialog.h"
 #include "ui_PreferencesDialog.h"
 
+#include "PreferencesKeys.h"
+#include "main.h"
+
+#include <QPushButton>
+#include <QSettings>
 #include <QStyleFactory>
 
 PreferencesDialog::PreferencesDialog(QWidget *parent) : QDialog(parent), ui(new Ui::PreferencesDialog) {
   ui->setupUi(this);
   this->setWindowFlag(Qt::WindowContextHelpButtonHint, false);
 
+  foreach (QString styleName, QStyleFactory::keys()) { ui->styleCombo->addItem(styleName); }
+
+  this->reset();
+
+  connect(ui->buttonBox->button(QDialogButtonBox::RestoreDefaults), &QAbstractButton::clicked, this,
+          &PreferencesDialog::restoreDefaultsClicked);
+  connect(ui->buttonBox->button(QDialogButtonBox::Apply), &QAbstractButton::clicked, this,
+          &PreferencesDialog::applyClicked);
+}
+
+PreferencesDialog::~PreferencesDialog() { delete ui; }
+
+void PreferencesDialog::apply() {
+  QSettings settings;
+  settings.beginGroup(preferencesKey());

```

Comment:
I don't understand the point of this `begin`/`endGroup` dynamic Qt seems to like, but I really appreciate how you've commented the ends to let readers know what preference they're editing. As discussed over Discord, I think it'd be good if there were some way to recursively reflect into the UI and do this procedural information gathering automatically. But that's an exercise for another time.